python/foglamp/common/process.py

```diff
@@ -73,51 +73,43 @@ def __init__(self):
         --port [core microservice management port]
         --name [process name]
         """
-        
+
         self._start_time = time.time()
 
-        try:    
-            self._core_management_host = self.get_arg_value("--address")
-            self._core_management_port = self.get_arg_value("--port")
-            self._name = self.get_arg_value("--name")
-        except ArgumentParserError:
+        try:
+            parser = SilentArgParse()
+            parser.add_argument("--name", required=True)
+            parser.add_argument("--address", required=True)
+            parser.add_argument("--port", required=True, type=int)
```

Comment:
Can we force these to be unique? and No duplicate append

Reply:
Need to investigate. But how do we decide which value to keep?

Reply:
when I say unique, I mean for required args only; and that should be the value supplied by Scheduler only, not by external task / process and protect our internal required args, if supplied (tried to override, reject and log)

Reply:
ArgumentParser.parse_known_args() retains only the last value if an argument is repeated. So effectively you are asking is to override/extend parse_know_args() method to add this functionality. Let us take this separately vide another ticket.

Reply:
Create a North task as under:
`north,"[""tasks/north"", ""--foo=FOO"", ""--bar=BAR"", ""--address=some""]"
`

It runs the process with arguments as:
`['tasks/north', '--foo=FOO', '--bar=BAR', '--address=some', '--port=38281', '--address=127.0.0.1', '--name=pi']
`
 
And syslog entries appear as:
```
Feb 15 15:34:55 asinha-ThinkPad-E450 FogLAMP[5764] INFO: scheduler: foglamp.services.core.scheduler.scheduler: Process started: Schedule 'pi' process 'north' task 164a7960-d701-48dc-82c5-507ad8c36985 pid 6737, 2 running tasks#012['tasks/north', '--foo=FOO', '--bar=BAR', '--address=some', '--port=38281', '--address=127.0.0.1', '--name=pi']
Feb 15 15:34:55 asinha-ThinkPad-E450 FogLAMP[6738] INFO: sending_process: sending_process_pi: Started

```

clearly indicating that there is no scope of Scheduler supplied arguments to be over written by some arbitrarily supplied values because Scheduler supplied arguments are appended at the end and which are retained as explained above.